CMSgov/mint-app
# [EASI-2187] Refactor DB Seeding to use resolver functions instead of .sql dump

# EASI-2187

## Changes and Description

- Modify `scripts/dev` to no longer reference `pgdump` or `psql` for dev data
- Create `cmd/dbseed`, which lightly wraps `package resolver` code to create development data more easily.

The main goal behind this change was to make it easier for new changes to implement seeded data, while also giving more visibility into _what_ seeded data is being created.

## Notes for reviewers:
- This PR does _not_ fully implement every single kind of seeded data that our old DB dump had. Notably, it does not seed documents. This is likely to be far easier with EASI-2205
- This PR does _not_ implement as many variations of Model Plans as we previously had. The intent of this PR was to set up a paradigm that allows us to easily add and edit our seeded data, so if you work on a ticket that could seed some data, follow the patterns in this code to enable more kinds of seeded data! (i.e. seeding more task list sections than just `Plan Basics`)

## How to test this change

1. Start the application with `scripts/dev up`
2. Run the Cypress tests with `yarn cypress run`
  - Ensure you have followed [the instructions for setting up Cypress & Okta auth locally](https://github.com/CMSgov/mint-app/blob/0b35902a1f707477716706b5d7bb68428b26b4eb/docs/local_testing.md#cypress-tests-end-to-end-integration-tests), otherwise `cypress/e2e/login.spec.js` will fail!
3. Run `scripts/dev db:clean`
4. Open the UI
5. Log in as `MINT` (using local auth)
6. Note there is no data!
7. Run `scripts/dev db:seed`
8. Refresh the page
9. Note that there's now seeded data!

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
